src/halmos/solve.py

```diff
@@ -61,7 +61,7 @@ class ModelVariable:
 halmos_var_pattern = re.compile(
     r"""
     \(\s*define-fun\s+               # Match "(define-fun"
-    \|?((?:halmos_|p_)[^ |]+)\|?\s+  # Capture either halmos_* or p_*, optionally wrapped in "|"
+    \|?((?:halmos_|p_)[^ |]+|storage_[^ |]+_00)\|?\s+  # Capture halmos_* or p_* or storage_*_00 optionally wrapped in "|"
```

Comment:
could you make including storage variables optional behind the --print-full-model flag?

for context: originally, storage variables used to be printed only when --print-full-model was enabled. that feature was somehow dropped in #437 (see the diff in `__main__.py`), and hasn't been brought back yet.